builtin/source/tabpage.ts

```diff
@@ -0,0 +1,109 @@
+import * as fn from "@denops/std/function";
+
+import { defineSource, type Source } from "../../source.ts";
+
+type Detail = {
+  /**
+   * Tab page number (1-indexed)
+   */
+  tabnr: number;
+
+  /**
+   * Window count in the tab page
+   */
+  windowCount: number;
+
+  /**
+   * Tab page variables
+   */
+  variables: Record<string, unknown>;
+
+  /**
+   * Windows in the tab page
+   */
+  windows: fn.WinInfo[];
+};
+
+export type TabpageOptions = {
+  /**
+   * Whether to include the tab label.
+   * If true, uses the tab label if set, otherwise shows buffer names.
+   * @default true
+   */
+  showLabel?: boolean;
+};
+
+/**
+ * Creates a Source that generates items from Vim tab pages.
+ *
+ * This Source retrieves information about all tab pages and generates items
+ * for each tab with details about windows and buffers in that tab.
+ *
+ * @param options - Options to customize tabpage listing.
+ * @returns A Source that generates items representing tab pages.
+ */
+export function tabpage(
+  options: Readonly<TabpageOptions> = {},
+): Source<Detail> {
+  const showLabel = options.showLabel ?? true;
+  return defineSource(async function* (denops, _params, { signal }) {
+    const currentTabnr = await fn.tabpagenr(denops);
+    const lastTabnr = await fn.tabpagenr(denops, "$");
+    const allWininfos = await fn.getwininfo(denops);
+    signal?.throwIfAborted();
+
+    const items = [];
+    for (let tabnr = 1; tabnr <= lastTabnr; tabnr++) {
+      // Get windows in this tab
+      const windows = allWininfos.filter((w) => w.tabnr === tabnr);
+      const windowCount = windows.length;
+
+      // Get tab variables
+      const variables = await fn.gettabvar(denops, tabnr, "&") as Record<
+        string,
+        unknown
+      >;
+
+      let label: string;
+      if (showLabel) {
+        // Try to get custom tab label
+        const customLabel = await fn.gettabvar(denops, tabnr, "label") as
+          | string
+          | null;
+        if (customLabel) {
+          label = customLabel;
+        } else {
+          // Create label from buffer names in the tab
+          const bufferNames = await Promise.all(
+            windows.map(async (w) => {
+              const name = await fn.bufname(denops, w.bufnr);
+              return name ? name.split("/").pop() || name : "[No Name]";
+            }),
+          );
+          label = bufferNames.join(", ");
+        }
+      } else {
+        label = `Tab ${tabnr}`;
+      }
+
+      // Add current tab indicator
+      const prefix = tabnr === currentTabnr ? "▸ " : "  ";
```

Comment:
It would be better to make the indicator customizable and ASCII character is preferred for the detaul value.